### ISSUE TITLE
Required vs urlParams

### ISSUE THREAD

**Issue by MarkMurphy** [stripe/stripe-node#514]
Looking at the source code I'm a little confused about `StripeMethod`s api. It doesn't document `urlParams` but expects that as part of `spec`. It also _**does**_ document `required` but doesn't appear to use it.

**Comment by remi-stripe** [comment]
@MarkMurphy The API ref mention that the parent id is required even though it is part of the URL. So for example to retrieve the card card_abc on customer cus_123, you hit the url `/v1/customers/cus_123/sources/card_abc` where `customer` and `id` are required (and part of the URL).

In stripe-node it looks like this:

```
stripe.customers.retrieveCard(
  "cus_123",
  "card_ABC",
  function(err, card) {
    // asynchronously called
  }
);
```

Let me know if this doesn't help!

**Comment by MarkMurphy** [comment]
@remi-stripe Apologies, I was referring to the internals of the stripe-node library. 

ie. https://github.com/stripe/stripe-node/blob/master/lib/StripeMethod.js#L13
The above method does not document `urlParams` but it is used [here](https://github.com/stripe/stripe-node/blob/master/lib/makeRequest.js#L11):

Also, what does this do: https://github.com/stripe/stripe-node/blob/88055dff171ec1d6cbcd43f5e5cadcabb28e3ace/lib/resources/Products.js#L16

`required` doesn't appear to be used anywhere.

**Comment by ob-stripe** [comment]
Hey @MarkMurphy, sorry for the late reply. You are correct: `required` should really be `urlParams` in the documentation. `required` doesn't actually do anything. I've pushed a fix in #516.
